<launch>
  
  <!-- <arg name="image" default="$(find vs_project)/goal_pose1.png"/> -->
  <arg name="image" default="$(find vs_project)/parking2.png"/>

  <!-- Publishes the image file of the robot in the goal position onto a rostopic -->
  <node pkg="image_publisher" type="image_publisher" name="image_publisher"
        args="$(arg image)" >
    <param name="flip_horizontal" value="false" />
    <param name="flip_vertical" value="false" />
    <param name="frame_id" value="my_camera" />
    <param name="publish_rate" value="1" />
    <param name="camera_info_url" value="file:///$(find vs_project)/cal_par_undistorted.yaml" />   <!-- relative to ~/.ros/ -->
  </node>
  
  <!-- Setting the path of the camera calibration file -->
  <arg name="yaml_file" default="$(find vs_project)/cal_par_undistorted.yaml"/>
  <!-- Setting the topic name where we get the camera images-->
  <arg name="camera" default="/image_publisher/image_raw"/>

  <param name="yaml_file" value="$(arg yaml_file)"/>
  <param name="camera" value="$(arg camera)"/>

  <remap from="current_position" to="goal_position" />
  <remap from="current_homo_matrix" to="goal_homo_matrix" />
  <remap from="image_aruco_detected" to="goal_aruco_detected" />
  <node pkg="vs_project" type="pose_estimator.py" name="pose_estimator" output="screen"></node>

  <remap from="image" to="/goal_aruco_detected" />
  <node name="image_view" pkg="image_view" type="image_view" respawn="true" output="screen"></node>
</launch>